api/types/kustomization.go

```diff
@@ -7,6 +7,14 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
```

Comment:
I'd prefer to not expand the deps of the `types` package to include  `filesys` and `kio`.

`types` is supposed to have very few deps and is already getting too big.

Suggest keeping `ConvertVarsToReplacements` in a new file in `kustomize/kustomize/commands/edit/fix`

maybe just  called `convert.go`, with an accompanying `convert_test.go`.
The `fix` package already depends on filesys, so no problem there.

Reply:
Question: should I move all this code to `kustomize/kustomize/commands/edit/fix` or just part of it? I'll run into the small problem of not being able to use `api/internal/utils.StringSliceContains` (because it's internal).

Reply:
done

Reply:
question: should `fix_vars_test.go` be renamed to `convert_test.go` or did you have something else in mind?